tests/perf/perf_tests.cc

```diff
@@ -185,6 +185,18 @@ static inline std::ostream& operator<<(std::ostream& os, duration d)
 
 }
 
+#if FMT_VERSION >= 90000
+
+}}
+
+template<>
+struct fmt::formatter<perf_tests::internal::duration> : fmt::ostream_formatter{};
+
+namespace perf_tests {
+namespace internal {
+
+#endif
```

Comment:
The namespace split and differentiation depending on the FMT_VERSION doesn't look nice.
Please, split the patch. First patch is to de-anonymize the duration class unconditionally. Second patch to add formatter for the internal::duration.